qutebrowser/browser/hints.py

```diff
@@ -454,21 +454,13 @@ def _hint_scattered(self, min_chars, chars, elems):
         # Determine how many digits the link hints will require in the worst
         # case. Usually we do not need all of these digits for every link
         # single hint, so we can show shorter hints for a few of the links.
-        needed = max(min_chars, math.ceil(math.log(len(elems), len(chars))))
+        needed = max(min_chars, math.ceil(
+            math.log(len(elems), len(chars))-1e-13))
         # Short hints are the number of hints we can possibly show which are
         # (needed - 1) digits in length.
-        if needed > min_chars:
+        if needed > min_chars and needed > 1:
             total_space = len(chars) ** needed
-            # Calculate short_count naively, by finding the avaiable space and
-            # dividing by the number of spots we would loose by adding a
-            # short element
-            short_count = math.floor((total_space - len(elems)) /
-                                     len(chars))
-            # Check if we double counted above to warrant another short_count
-            # https://github.com/qutebrowser/qutebrowser/issues/3242
-            if total_space - (short_count * len(chars) +
-                              (len(elems) - short_count)) >= len(chars) - 1:
-                short_count += 1
+            short_count = (total_space - len(elems)) // (len(chars) - 1)
```

Comment:
@jgkamat  About "`math.log` being slow":

* That part is not performance critical. For large number of links, it takes much longer to display the hint labels than to compute the hint strings.
* I decide to time it to see if it's really slower. [Result](https://tio.run/##fZDfboMgFMbveQrukI4uomsvmvIcu2yo0kkiaAC7mqbP7gB1tmaZF@Z8h/M7f762d1Wj82G4mEZBJ5WQDkrVNsZNCkxKcVeBUlxg3XydTNPpUpSnrk1ELZQlsKi4sfgAoP@uvIYM0hjzouiUV/E9Zr4rWYspf4QRH7EZfZvZhd88NzDCdUaHUhAXcsK6pCe3ML01Urvkgnp27x/kxu63B8IAOsrGa5Kaq3PJD@Ga90LIOomRvyl22FKxpTmGRHfqLAyjaZoGPFvhKw8Cuma4tuz/Kcu2ofbuf3FXbq3wfocc@2POTKFPqbUwB4gImrsjKL0d9Ogy76sVEL3y6BfGAETXaLYjOzzG@6c4nx6mKn8RoSle1FqOekxk@80mJ9keL/LjVe6CHIYf): For results larger than 3 or 4 `math.log` is faster, otherwise `log_rounded_up` is faster.

Reply:
Fixed.

It's pretty easy, anyway, so I don't think it's very necessary.